sale_order_invoicing_finished_task/models/project.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2017 Camptocamp SA
+# Copyright 2017 Tecnativa - Sergio Teruel
+# Copyright 2017 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProjectTaskType(models.Model):
+    _inherit = "project.task.type"
+
+    invoiceable = fields.Boolean()
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    invoiceable = fields.Boolean()
+    invoicing_finished_task = fields.Boolean(
+        related="sale_line_id.product_id.invoicing_finished_task",
+    )
+
+    @api.onchange("stage_id")
+    def _onchange_stage_id(self):
+        tasks = self.filtered(
+            lambda t: (
+                t.invoicing_finished_task
+                and t.stage_id.invoiceable
+                and not t.invoiceable
+            )
+        )
+        tasks.toggle_invoiceable()
+
+    def toggle_invoiceable(self):
+        self._check_sale_line_state()
+        for task in self:
+            task.invoiceable = not task.invoiceable
+
+    def write(self, vals):
+        if "sale_line_id" in vals:
+            self._check_sale_line_state(vals["sale_line_id"])
+        res = super().write(vals)
+        if "invoiceable" in vals:
+            self.mapped("sale_line_id")._compute_qty_delivered()
```

Comment:
should this not be added to the depends of said compute?

Reply:
Done